src/NativeShell/src/media/RemotePlayerService.java

```diff
@@ -232,97 +273,73 @@ private void setMediaSessionMetadata(MediaSession mediaSession, String itemId, S
             if (largeIcon != null) {
                 metadataBuilder.putBitmap(MediaMetadata.METADATA_KEY_ALBUM_ART, largeIcon);
             }
-
             mediaSession.setMetadata(metadataBuilder.build());
         }
     }
 
     private void onStopped() {
         NotificationManager notificationManager = (NotificationManager) getApplicationContext().getSystemService(Context.NOTIFICATION_SERVICE);
-        notificationManager.cancel(1);
+        notificationManager.cancel(notifyId);
         Intent intent = new Intent(getApplicationContext(), RemotePlayerService.class);
         stopService(intent);
     }
 
     private Notification.Action generateAction(int icon, String title, String intentAction) {
-
         Intent intent = new Intent(getApplicationContext(), RemotePlayerService.class);
         intent.setAction(intentAction);
-        PendingIntent pendingIntent = PendingIntent.getService(getApplicationContext(), 1, intent, 0);
-
+        PendingIntent pendingIntent = PendingIntent.getService(getApplicationContext(), notifyId, intent, 0);
         return new Notification.Action(icon, title, pendingIntent);
     }
 
-    @Override
-    public int onStartCommand(final Intent intent, int flags, int startId) {
-        if (mediaSessionManager == null) {
-            initMediaSessions();
-        }
-
-        handleIntent(intent);
-        return super.onStartCommand(intent, flags, startId);
-    }
-
     private void initMediaSessions() {
-        mediaSessionMediaId = "";
-        largeItemIcon = null;
-
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             mediaSessionManager = (MediaSessionManager) getSystemService(Context.MEDIA_SESSION_SERVICE);
-            mediaSession = new MediaSession(getApplicationContext(), "sample session");
+            mediaSession = new MediaSession(getApplicationContext(), getClass().toString());
             mediaController = new MediaController(getApplicationContext(), mediaSession.getSessionToken());
-            mediaSession.setActive(true);
-            int flags = MediaSessionCompat.FLAG_HANDLES_TRANSPORT_CONTROLS | MediaSessionCompat.FLAG_HANDLES_MEDIA_BUTTONS;
-            mediaSession.setFlags(flags);
 
+            mediaSession.setActive(true);
+            mediaSession.setFlags(MediaSession.FLAG_HANDLES_TRANSPORT_CONTROLS | MediaSession.FLAG_HANDLES_MEDIA_BUTTONS);
             mediaSession.setCallback(new Callback() {
                 @Override
                 public void onPlay() {
                     super.onPlay();
-                    Log.i(Constants.LOG_TAG, "onPlay");
```

Comment:
why drop logging?

Reply:
Logging needs an overhaul which will have to start in the Java API client first, plus this is already logged once on the WebView side, plug logging isn't even working in Cordova at the moment.